simulation/parser.go

```diff
@@ -143,24 +158,38 @@ func parseAdversaryConfig(adversaryDelays *string, adversaryTypes *string, adver
 	if *adversaryMana != "" {
 		config.AdversaryMana = parseStrToFloat64(*adversaryMana)
 	}
-
-	// no adversary if simulation target is not DS
-	if config.SimulationTarget != "DS" {
-		config.AdversaryTypes = []int{}
+	if *adversaryNodeCounts != "" {
+		config.AdversaryNodeCounts = parseStrToInt(*adversaryNodeCounts)
+	}
+	if *adversaryInitColors != "" {
+		config.AdversaryInitColors = parseStr(*adversaryInitColors)
 	}
 
-	// maximum 3 adversary types allowed for one simulation, one per color
-	if len(config.AdversaryTypes) > 3 {
+	// no adversary if colors are not provided
+	if len(config.AdversaryInitColors) != len(config.AdversaryTypes) {
 		config.AdversaryTypes = []int{}
 	}
 
-	// make sure mana and delays are only defined when adversary type is provided and have the same length
+	// make sure mana, nodeCounts and delays are only defined when adversary type is provided and have the same length
 	if len(config.AdversaryDelays) != 0 && len(config.AdversaryDelays) != len(config.AdversaryTypes) {
```

Comment:
```suggestion
	if len(config.AdversaryDelays) != 0 && len(config.AdversaryDelays) != len(config.AdversaryTypes) {
		log.Warnf("The AdversaryDelays count is not equal to AdversaryTypes count!")	
```
We can use warning log to warn the user the setting might be wrong.